### ISSUE TITLE
Instructions to build Roslyn has issue with restore.cmd

### ISSUE THREAD

**Issue by paul1956** [dotnet/roslyn#7226]
The line below requires knowledge not provide
```
Run  Restore.cmd  in the command prompt to restore NuGet packages
```
There is no Restore.cmd understood by command.exe, if you mean "NuGet.exe restore" then you need to specify a solution file because there are several .sln files in the directory. Do you run the NuGet command for all solutions, does order matter? It would help to just give the commands I can type, there is less confusion that way.

There are also warnings
```
C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets
(3962,5): warning MSB3026: Could not copy "C:\Users\Paul\.nuget\packages\Micros
oft.CodeAnalysis.Test.Resources.Proprietary\1.1.0-beta1-20150824-02\lib\net45\M
icrosoft.CodeAnalysis.Test.Resources.Proprietary.dll" to "C:\Users\Paul\Documen
ts\GitHub\repos\roslyn-update-1\Binaries\Debug\Microsoft.CodeAnalysis.Test.Reso
urces.Proprietary.dll". Beginning retry 1 in 1000ms. The process cannot access
the file 'C:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\Binaries\Debug\M
icrosoft.CodeAnalysis.Test.Resources.Proprietary.dll' because it is being used
by another process. [C:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\src\T
est\Utilities\Portable\TestUtilities.csproj]
```
and errors
```
C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets(109,5):
error : Your project.json doesn't list 'win7' as a targeted runtime. You should
 add '"win7": { }' inside your "runtimes" section in your project.json, and the
n re-run NuGet restore. [C:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\s
rc\Samples\CSharp\ConsoleClassifier\ConsoleClassifierCS.csproj]
C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets(109,5):
error : Your project.json doesn't have a runtimes section. You should add '"run
times": { "win7": { } }' to your project.json and then re-run NuGet restore. [C
:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\src\Samples\CSharp\APISampl
eUnitTests\APISampleUnitTestsCS.csproj]
C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets(109,5):
error : Your project.json doesn't have a runtimes section. You should add '"run
times": { "win7": { } }' to your project.json and then re-run NuGet restore. [C
:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\src\Samples\CSharp\ConvertT
oAutoProperty\Impl\ConvertToAutoPropertyCS.csproj]
C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets(109,5):
error : Your project.json doesn't have a runtimes section. You should add '"run
times": { "win7": { } }' to your project.json and then re-run NuGet restore. [C
:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\src\Samples\CSharp\Implemen
tNotifyPropertyChanged\Impl\ImplementNotifyPropertyChangedCS.csproj]
C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets(109,5):
error : Your project.json doesn't have a runtimes section. You should add '"run
times": { "win7": { } }' to your project.json and then re-run NuGet restore. [C
:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\src\Samples\VisualBasic\API
SampleUnitTests\APISampleUnitTestsVB.vbproj]
C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets(109,5):
error : Your project.json doesn't have a runtimes section. You should add '"run
times": { "win7": { } }' to your project.json and then re-run NuGet restore. [C
:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\src\Samples\Shared\UnitTest
Framework\UnitTestFramework.csproj]
C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets(109,5):
error : Your project.json doesn't list 'win7' as a targeted runtime. You should
 add '"win7": { }' inside your "runtimes" section in your project.json, and the
n re-run NuGet restore. [C:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\s
rc\Samples\VisualBasic\FormatSolution\FormatSolutionVB.vbproj]
C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets(109,5):
error : Your project.json doesn't have a runtimes section. You should add '"run
times": { "win7": { } }' to your project.json and then re-run NuGet restore. [C
:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\src\Samples\VisualBasic\Con
vertToAutoProperty\Impl\ConvertToAutoPropertyVB.vbproj]
C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets(109,5):
error : Your project.json doesn't list 'win7' as a targeted runtime. You should
 add '"win7": { }' inside your "runtimes" section in your project.json, and the
n re-run NuGet restore. [C:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\s
rc\Samples\VisualBasic\ConsoleClassifier\ConsoleClassifierVB.vbproj]
C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets(109,5):
error : Your project.json doesn't have a runtimes section. You should add '"run
times": { "win7": { } }' to your project.json and then re-run NuGet restore. [C
:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\src\Samples\VisualBasic\Mak
eConst\Impl\MakeConstVB.vbproj]
C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets(109,5):
error : Your project.json doesn't have a runtimes section. You should add '"run
times": { "win7": { } }' to your project.json and then re-run NuGet restore. [C
:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\src\Samples\VisualBasic\Imp
lementNotifyPropertyChanged\Impl\ImplementNotifyPropertyChangedVB.vbproj]
C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets(109,5):
error : Your project.json doesn't have a runtimes section. You should add '"run
times": { "win7": { } }' to your project.json and then re-run NuGet restore. [C
:\Users\Paul\Documents\GitHub\repos\roslyn-update-1\src\Samples\CSharp\MakeCons
t\Impl\MakeConstCS.csproj]
```
Are these expected?

**Comment by leppie** [comment]
*There is no Restore.cmd understood by command.exe* 

I have no idea what you mean there. What is command.exe? Do you mean cmd.exe? Batch files will not run in a bash shell from what I know.

You just open the Developer Command Prompt for VS2015, CD to roslyn dir, and run Restore.cmd that is there. That is it.

**Comment by paul1956** [comment]
In the update-1 directory there is no restore.cmd

**Comment by leppie** [comment]
Ahh, I see what you mean, the `update-1` branch does not have `Restore.cmd`. It was only create just before November. https://github.com/dotnet/roslyn/commits/master/Restore.cmd Perhaps you can just copy it into the tree.

**Comment by davkean** [comment]
@jasonmalinowski should we port Restore.cmd to Update-1?

**Comment by jasonmalinowski** [comment]
Ah, crap, yep, we should do that. My bad for not fixing that earlier.

**Comment by diryboy** [comment]
For your warning messages, I hit that a few times, it seems to me that killing any VSHub.exe MSBuild.exe and VBCSCompiler.exe before doing it solved the problem.

**Comment by davkean** [comment]
The warnings are due to some flaky build issues that we're trying to chase down, I filed https://github.com/dotnet/roslyn/issues/7248 to track the warning.

**Comment by diryboy** [comment]
@davkean Oh really? Maybe I should have reported, mine dll was different, it was something like core-clr/Microsoft.CodeAnalysis.dll, but I'm not hitting this recently.

**Comment by davkean** [comment]
@diryboy The one above is due to flaky builds. The one you are hitting is probably because of tests - we have compiler server tests that are leaving instances of vbcscompiler open. It's known, and we're trying to fix it.

**Comment by diryboy** [comment]
@davkean Thanks for the info.

**Comment by davkean** [comment]
@diryboy Please do report issues that you encounter in day-to-day development - we're trying to make sure we capture all the issues, and improve the overall developer experience.

**Comment by jasonmalinowski** [comment]
And as far as getting Restore.cmd into update-1, there is now PR #7249 to do exactly that.

**Comment by davkean** [comment]
This was resolved in #7249.
